compiler/codegen/expr.cpp

```diff
@@ -4952,53 +4957,49 @@ DEFINE_PRIM(PRIM_LOOKUP_FILENAME) {
     ret = call->codegenBasicPrimitiveExpr();
 }
 
-
-GenRet CallExpr::codegenPrimitive() {
-  SET_LINENO(this);
-
-  static bool codegenPrimitivesRegistered = false;
-  if (!codegenPrimitivesRegistered) {
```

Comment:
Was the problem with the original approach just that nothing ever set `codegenPrimitivesRegistered = true`?

Reply:
Looks like it - maybe you should have reviewed it :P